test/ocsp/helper/helper.go

```diff
@@ -227,17 +270,20 @@ func checkSignerTimes(resp *ocsp.Response, issuer *x509.Certificate) error {
 	return nil
 }
 
-func parseAndPrint(respBytes []byte, cert, issuer *x509.Certificate, expectStatus int) (*ocsp.Response, error) {
+func parseAndPrint(respBytes []byte, cert, issuer *x509.Certificate, config CheckOCSPConfig) (*ocsp.Response, error) {
 	fmt.Printf("\nDecoding body: %s\n", base64.StdEncoding.EncodeToString(respBytes))
 	resp, err := ocsp.ParseResponseForCert(respBytes, cert, issuer)
 	if err != nil {
 		return nil, fmt.Errorf("parsing response: %s", err)
 	}
-	if resp.Status != expectStatus {
-		return nil, fmt.Errorf("wrong CertStatus %d, expected %d", resp.Status, expectStatus)
+	if config.expectStatus != -1 && resp.Status != config.expectStatus {
+		return nil, fmt.Errorf("wrong CertStatus %d, expected %d", resp.Status, config.expectStatus)
+	}
+	if config.expectReason != -1 && resp.RevocationReason != config.expectReason {
+		return nil, fmt.Errorf("wrong RevocationReason %d, expected %d", resp.RevocationReason, config.expectReason)
```

Comment:
I left a comment about this in #4885, but since you're changing this and adding more checks, I really think the behavior would be much better if the tool always printed the response contents and *then* made the assertions it wants. This would let you move the asserts out of this function too and thus reduce it's reliance on `CheckOCSPConfig`.

Reply:
I actually have an idea for a slightly more thorough way to do this, lemme do it in a follow-up change.

Reply:
Follow-up PR here: https://github.com/letsencrypt/boulder/pull/4910